### ISSUE TITLE
Crate doesn't compile on ARM

### ISSUE THREAD

**Issue by petevine** [sfackler/rust-openssl#321]
Build fails on arm linux using stable rust 1.5.0:

````
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:92:50: 92:64 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:92             try_ssl!(ffi::BN_dec2bn(v.raw_ptr(), c_str.as_ptr()));
                                                                                                                                                 ^~~~~~~~~~~~~~
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/macros.rs:24:19: 24:39 note: in this expansion of try_ssl_if! (defined in /home/odroid/.cargo-1.5054423464/openssl-0.7.1/src/macros.rs)
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:92:13: 92:67 note: in this expansion of try_ssl! (defined in /home/odroid/.cargo-1.5/re423464/openssl-0.7.1/src/macros.rs)
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:92:50: 92:64 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:100:50: 100:64 error: mismatched types:
 expected `*const u8`,
    found `*const i8`
(expected u8,
    found i8) [E0308]
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:100             try_ssl!(ffi::BN_hex2bn(v.raw_ptr(), c_str.as_ptr()));
                                                                                                                                                  ^~~~~~~~~~~~~~
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/macros.rs:24:19: 24:39 note: in this expansion of try_ssl_if! (defined in /home/odroid/.cargo-1.5054423464/openssl-0.7.1/src/macros.rs)
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:100:13: 100:67 note: in this expansion of try_ssl! (defined in /home/odroid/.cargo-1.5/54423464/openssl-0.7.1/src/macros.rs)
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:100:50: 100:64 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:424:56: 424:59 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:424             let str = String::from_utf8(CStr::from_ptr(buf).to_bytes().to_vec()).un
                                                                                                                                                        ^~~
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:424:56: 424:59 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:434:56: 434:59 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:434             let str = String::from_utf8(CStr::from_ptr(buf).to_bytes().to_vec()).un
                                                                                                                                                        ^~~
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/bn/mod.rs:434:56: 434:59 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/ssl/error.rs:120:36: 120:66 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/ssl/error.rs:120         let bytes = CStr::from_ptr(ffi::ERR_lib_error_string(err)).to_bytes().to
                                                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/ssl/error.rs:120:36: 120:66 help: run `rustc --explain E0308` to see a detailed explanation
/home/odroid/.cargo-1.5/registry/src/github.com-48ad6e4054423464/openssl-0.7.1/src/ssl/error.rs:127:36: 127:67 error: mismatched types:
 expected `*const i8`,
    found `*const u8`
(expected i8,
    found u8) [E0308]
````

and so on. This is somewhat related to https://github.com/rust-lang/rust/issues/29867 (through libc 0.2 update) but the crate actually compiles on 1.6!

**Comment by sfackler** [comment]
I believe this has been fixed on master but I haven't yet released it (oops!). Can you confirm that this commit builds for you? https://github.com/sfackler/rust-openssl/commit/f54af75eb7f8c861383119722f548cb2866ca813. I'll push a release if so.

**Comment by petevine** [comment]
Yes, it does.

**Comment by sfackler** [comment]
Cool, I will push a release out tonight.

**Comment by petevine** [comment]
Great, thx.

**Comment by sfackler** [comment]
Released v0.7.2
